core-rust/state-manager/src/jni/rust_global_context.rs

```diff
@@ -159,31 +161,38 @@ pub struct StateManagerConfig {
     pub no_fees: bool,
 }
 
-pub type ActualStateManager = StateManager<StateManagerDatabase>;
+impl RadixNodeConfig {
+    pub fn new_for_testing() -> Self {
+        RadixNodeConfig {
+            network_definition: NetworkDefinition::simulator(),
+            mempool_config: Some(MempoolConfig::new_for_testing()),
+            vertex_limits_config: None,
+            database_backend_config: DatabaseBackendConfig::InMemory,
+            database_flags: DatabaseFlags::default(),
+            logging_config: LoggingConfig::default(),
+            no_fees: false,
+        }
+    }
+}
 
-pub struct JNIStateManager {
-    pub runtime: Arc<Runtime>,
-    pub network: NetworkDefinition,
-    pub state_manager: Arc<ActualStateManager>,
+#[derive(Clone)]
+pub struct RadixNode {
+    pub state_manager: Arc<StateManager<StateManagerDatabase>>,
     pub database: Arc<RwLock<StateManagerDatabase>>,
     pub pending_transaction_result_cache: Arc<RwLock<PendingTransactionResultCache>>,
     pub mempool: Arc<RwLock<PriorityMempool>>,
     pub mempool_manager: Arc<MempoolManager>,
     pub committability_validator: Arc<CommittabilityValidator<StateManagerDatabase>>,
     pub transaction_previewer: Arc<TransactionPreviewer<StateManagerDatabase>>,
-    pub metric_registry: Arc<Registry>,
 }
 
-impl JNIStateManager {
-    pub fn init(env: &JNIEnv, j_state_manager: JObject, j_config: jbyteArray) {
-        let config_bytes: Vec<u8> = jni_jbytearray_to_vector(env, j_config).unwrap();
-        let config = StateManagerConfig::from_java(&config_bytes).unwrap();
-
-        let runtime = Runtime::new().unwrap();
-
-        setup_tracing(&runtime, std::env::var("JAEGER_AGENT_ENDPOINT").ok());
-
-        // Build the basic subcomponents.
+impl RadixNode {
```

Comment:
`NodeRustEnvironment` can not be moved outside of `state-manager` without moving almost everything from `jni` (including `core-api/jni`) because of the circular dependencies created.